node_name: "attitude_controller_node"
node_rate: 20

use_optical_flow_velocities: true  # true -> Using velocity estimates from optical_flow
                                # false -> Using polled velocity estimates 

use_pure_pursuit_guidance: false    # true -> Using pure pursuit guidance to generate the velocity reference
                                # false -> Using PID-guidance to generate the velocity reference

attitude_limits:
  pitch: [-5, 5]  # deg
  roll: [-5, 5]   # deg

velocity_reference_model:
  omegas: [10, 10]
  zetas: [1, 1]

  # First order reference model for vertical velocity
  T_z: 0.5
  K_z: 1

pid: 
  x_axis:
    kp: 0.25 #7 # 0.25 # -7 
    ki: 0.01 # 0.01 # 0.01   # -0.1
    kd: 0
  y_axis:
    kp: 0.25 # 7 # 0.25 # 7
    ki: 0.01 # 0.01 # 0.01   # 0.1
    kd: 0

linear_drag_model:
  drone_mass: 315
  g: 9.81
  dx: 0.08063504
  dy: 0.09929089

ipid:
  x_axis:
    alpha: 1
    kp: 7
    ki: 0.1
    kd: 0.5
  y_axis:
    alpha: 1
    kp: -7
    ki: -0.1
    kd: -0.5
